edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -513,3 +514,332 @@ def test_proctoring_providers_list_empty(self):
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            hide_after_due=False,
+            is_active=True
+        )
+
+        no_due_date_exam_id = no_due_date_exam.id
+        no_due_date_url = reverse('api:v1:exam-access-tickets', kwargs={'exam_id': no_due_date_exam_id})
+
+        start_time = timezone.now()
+        allowed_time_limit_mins = no_due_date_exam.time_limit_mins
+        ExamAttempt.objects.create(
+            user=self.user,
+            exam=no_due_date_exam,
+            attempt_number=1,
+            status='created',
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+        response = self.get_exam_access(self.user, no_due_date_url)
+        self.assertEqual(403, response.status_code)
+
+    def test_access_not_granted_not_verified_attempt(self):
+        """
+        Verify the endpoint does not grant access for past-due exam
+        with an exam attempt whose status is not verified.
+        """
+        start_time = self.past_due_date - timedelta(minutes=60)
+        allowed_time_limit_mins = self.past_due_exam.time_limit_mins
+        ExamAttempt.objects.create(
+            user=self.user,
+            exam=self.past_due_exam,
+            attempt_number=1,
+            status='started',
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+        response = self.get_exam_access(self.user, self.past_due_url)
+        self.assertEqual(403, response.status_code)
+
+    def test_access_not_granted_if_hide_after_due(self):
+        """
+        Verify the endpoint does not grant access for past-due exam
+        when the exam is set to hide after due.
+        """
+        past_due_exam_hide = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id='abcd1234',
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date=self.past_due_date,
+            hide_after_due=True,
+            is_active=True
+        )
+
+        past_due_exam_id = past_due_exam_hide.id
+        past_due_url = reverse('api:v1:exam-access-tickets', kwargs={'exam_id': past_due_exam_id})
+
+        start_time = self.past_due_date - timedelta(minutes=60)
+        allowed_time_limit_mins = past_due_exam_hide.time_limit_mins
+        ExamAttempt.objects.create(
+            user=self.user,
+            exam=past_due_exam_hide,
+            attempt_number=1,
+            status='verified',
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+        response = self.get_exam_access(self.user, past_due_url)
+        self.assertEqual(403, response.status_code)
+
+    def test_access_not_granted_not_started_exam_attempt(self):
```

Comment:
no need to refactor unless you want to but most of these tests are performing the same logic but with different inputs for attempt status, past due, and expected response. You can use [ddt](https://ddt.readthedocs.io/en/latest/) to test a bunch of these all at once.

Take a look at usage in edx-proctoring as an example
https://github.com/openedx/edx-proctoring/blob/0ae746c253db5910cd6f91abf60e19771102130c/edx_proctoring/tests/test_api.py#L1876